x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/snapshot/upgrade/SnapshotUpgradeState.java

```diff
@@ -35,4 +37,15 @@ public String toString() {
         return name().toLowerCase(Locale.ROOT);
     }
 
+    /**
+     * @return {@code false} if state matches any of the given {@code candidates}
```

Comment:
```suggestion
     * @return {@code true} if state matches none of the given {@code candidates}
```

nit Documenting the true case often means fewer mental negations

Reply:
I fixed this in all our `*State.isNoneOf()` methods